### PR TITLE
First version

### PR DESCRIPTION
**Pull request by MiLk**
It should be possible to improve the installation/integration process, but the code should be good.

It requires https://github.com/mage/mage-sdk-cpp/pull/22

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+mkdir -p "${project_directory}/Assets/{Editor,Plugins}"
+
+for file in Editor/*.{cs,py}; do
+    cp ${file} "${project_directory}/Assets/Editor/"
+done
+
+sed "s#MAGE_SDK_CPP_PATH#${MAGE_SDK_CPP_PATH}#g" Editor/post_process.py.tpl > "${project_directory}/Assets/Editor/post_process.py"
+
+cp -R Plugins/ "${project_directory}/Plugins/"
```
Comment (MiLk) [inline scripts/build.sh]: `cp -R Plugins/ "${project_directory}/Assets/Plugins/"`

File: Plugins/MAGE/RPC.cs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                                               string mageProtocol) {
+            return IntPtr.Zero;
+        }
+        private static void MAGE_RPC_Disconnect(IntPtr client) {}
+        private static IntPtr MAGE_RPC_Call(IntPtr client, string methodName, string parameters, IntPtr ok) {
+            return IntPtr.Zero;
+        }
+        private static void MAGE_RPC_SetSession(IntPtr client, string sessionKey) {}
+        private static void MAGE_RPC_ClearSession(IntPtr client) {}
+        private static void MAGE_RPC_PullEvents(IntPtr client, int transport) {}
```
Comment (MiLk) [inline Plugins/MAGE/RPC.cs]: Should return an int.

File: Plugins/MAGE/RPC.cs [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+    public enum Transport {
+        SHORTPOLLING = 0,
+        LONGPOLLING
+    };
+
+    public class RPC {
+
+        // Import from libraries
+        #if UNITY_IPHONE
```
Comment (MiLk) [inline Plugins/MAGE/RPC.cs]: and ! UNITY_EDITOR

**Comment by nullorvoid** [comment]
I had a look through the PR, a few things that I noticed.
1. The use of If/else defs is a lot of separation that although doesn't separate functionality it separates meaning, I think that they are overused and could do with some sort of abstraction layer.  Nothing overly complex but maybe using factory pattern to generate the correct class for each situation.  You could then use the if/else to create the correct concrete class for mobile/android/etc.
2. Lack of linting, there are a few code style inconsistencies such as missing {} and spacing that I noticed throughout the file.  I have not specifically gone and highlighted each one as discussed we do not have a tool to do this, however I think we should look at running something over this code, either in xcode / monodevelop / Visual Studio depending on which development environment we use.

These were the major issues, I couldn't see any other problems with the code itself although I could do with another proper look through it.

The first point also applies to the C++ PR

**Comment by MiLk** [comment]
This PR can be closed, all the code lives in https://github.com/Wizcorp/ardritactics

**Pull request closed without merging**